lib/components/RunFrameWithApi/RunFrameWithApi.tsx

```diff
@@ -111,6 +119,28 @@ export const RunFrameWithApi = (props: RunFrameWithApiProps) => {
             ...ee,
           }),
         })
+
+        const manualEditsFilePath =
+          guessManualEditsFilePath(Array.from(fsMap.keys())) ??
+          "manual-edits.json"
+
+        const manualEditsFile = fsMap.get(manualEditsFilePath)
+
+        const updatedManualEdits = applyPcbEditEventsToManualEditsFile({
+          circuitJson: circuitJson!,
+          editEvents: [ee],
+          manualEditsFile: JSON.parse(manualEditsFile ?? "{}"),
+        })
+
+        fetch(`${API_BASE}/files/upsert`, {
+          method: "POST",
+          headers: { "Content-Type": "application/json" },
+          body: JSON.stringify({
+            file_path: manualEditsFilePath,
+            text_content: JSON.stringify(updatedManualEdits),
+            initiator: "filesystem_change",
```

Comment:
```suggestion
            initiator: "runframe",
```

Reply:
it's really important for this to be correct, otherwise will lead to infinite loops!